### ISSUE TITLE
Stack overflow reading type expression

### ISSUE THREAD

**Issue by dsheets** [lpw25/opam-doc-base#1]
``` sh
$ opam-doc-test-cmi ~/.opam/4.02.0+doc/lib/opam2web/o2wProject.cmi 
opam-doc-test-cmi: internal error, uncaught exception:
                   Stack overflow
```

```
Called from file "_build/lib-opam-doc-base/opamDocCmi.ml", line 134, characters 27-62
                   Called from file "list.ml", line 66, characters 22-25
```

Here: https://github.com/lpw25/opam-doc-base/blob/master/src/opamDocCmi.ml#L134

**Comment by lpw25** [comment]
I'm guessing that is a recursive type. I thought I'd avoided them by not supporting objects or polymorphic variants, but of course you can create a recursive type through a type constructor if that constructor is an alias for an object or polymorphic variant type. IIRC that is how cow represents html fragments, so that is probably what is going on here.

**Comment by dsheets** [comment]
http://erratique.ch/software/xmlm/doc/Xmlm#TYPEfrag

with `as 'a` appended. What is necessary to support this?

**Comment by dsheets** [comment]
Or at least fail gracefully.

**Comment by lpw25** [comment]
> Or at least fail gracefully.

I need to detect loops in types.

> What is necessary to support this?

I need to print the loops with an "as 'a" for them. The logic already exists in the compiler so it is not that hard to add. I could also fix variables while I'm there so that they are named correctly, which would be nice.

**Comment by lpw25** [comment]
Addressed in 0.1.2

**Comment by dsheets** [comment]
Cool! Thanks!
